File: Java/Jump-Game.java
public class Solution {
    /**
     * @param A: A list of integers
     * @return: The boolean answer
     */
    public boolean canJump(int[] A) {
        // wirte your code here
        int maxlen = 0;
        for (int i = 0; i <= maxlen && i < A.length; i ++) {
            maxlen = Math.max(maxlen, A[i] + i);
            if (maxlen >= A.length - 1) return true;
        }
        return false;
    }
}

File: Cpp/Jump-Game.cpp
#include <vector>

class Solution {
public:
    /**
     * @param A: A list of integers
     * @return: The boolean answer
     */
    bool canJump(vector<int> A) {
        // write you code here
        int cnt = A[0], n = A.size();
        for(int i = 1; i <= cnt; i++) {
            if (A[i] + i > cnt) cnt = A[i] + i;
            if (cnt >= n - 1) return true;
        }
        return false;
    }
};

